### DISCUSSION TITLE
How to schedule a job when using FlowProducer?

### DISCUSSION THREAD

**Discussion by apoorv569** [taskforcesh/bullmq#2855] [Q&A]
When using `FlowProducer` for defining jobs, how to configure them to run on a schedule, like every day at a certain time for example?

Like I have this,
```typescript
const directory_scan_queue = new Queue("directory_scan", queue_opts);
const metadata_extractor_queue = new Queue("metadata_extractor", queue_opts);
const save_metadata_to_db_queue = new Queue("save_metadata_to_db", queue_opts);
```

```typescript
const queues = [
    directory_scan_queue,
    metadata_extractor_queue,
    save_metadata_to_db_queue
];

for (const queue of queues) {
    queue.upsertJobScheduler(`${queue.name}_scheduler`, {
        every: 10000 // job will repeat every 10000 ms (10 sec)
    }, {
        opts: {
            backoff: 3,
            attempts: 5,
            removeOnFail: 1000,
        }
    });
}
```

```typescript
    const flow = new FlowProducer();

    flow.add({
        name: "save_metadata",
        queueName: "save_metadata_to_db",
        children: [
            {
                name: "extract_metadata",
                queueName: "metadata_extractor",
                children: [
                    {
                        name: "scan_directory",
                        queueName: "directory_scan",
                    }
                ]
            },
        ]
    });
```

But the `jobs` only run once still and never repeats (which in this case should be every 10s).

**Comment by roggervalf** [comment]
you can create a repeatable job that uses a flowProducer instance to add a flow

**Reply by apoorv569** [reply]
In the docs they mention this on this page https://docs.bullmq.io/guide/jobs/repeatable,
> Note: from BullMQ version 5.16.0 and onwards, we have deprecated these APIs in favor of ["Job Schedulers"](https://docs.bullmq.io/guide/job-schedulers), which provide a more cohesive and more robust API for handling repeatable jobs.

That's why I was using `JobSchedulers` instead which tells me do `queue.upsertJobScheduler()`, which I also forgot to `await`, but it then gives me a `Job`.

What do you mean by,
> you can create a repeatable job that uses a flowProducer instance to add a flow

Like I use this `Job` and assign it another `Worker` that just adds the `Flow` I have?
